Calculator/Calculator.cpp

```diff
@@ -20,24 +18,57 @@ class Token {
 
 //------------------------------------------------------------------------------
 
-Token get_token()    // read a token from cin
+class Token_stream {
+public:
+    Token_stream(); // make a Token_stream that reads from cin
+    Token get(); // get a Token
+    void putback(Token t); // put a Token back
+private:
+    bool full{ false }; // is there a Token in the buffer?
+    Token buffer; // here is where we keep a Token put back using putback()
+
+};
+
+
+//------------------------------------------------------------------------------
+
+// The constructor just sets full to indicate that the buffer is empty:
+Token_stream::Token_stream()
+    :full(false), buffer(0)    // no Token in buffer
 {
-    char ch;
-    cin >> ch;    // note that >> skips whitespace (space, newline, tab, etc.)
+}
 
+//------------------------------------------------------------------------------
+
+void Token_stream::putback(Token t)
+{
+    if (full) error("putback() into a full buffer");
+    buffer = t; // copy t to buffer
+    full = true; // buffer is now full
+}
+
+//------------------------------------------------------------------------------
+
+Token Token_stream::get()
+{
+    if (full) { // do we already have a Token ready?
+        full = false; // remove Token from buffer
+        return buffer;
+    }
+    char ch;
+    cin >> ch; // note that >> skips whitespace (space, newline, tab, etc.)
     switch (ch) {
-        //not yet   case ';':    // for "print"
-        //not yet   case 'q':    // for "quit"
+    case '=': // for �print�
+    case 'x': // for �quit�
     case '(': case ')': case '+': case '-': case '*': case '/':
-        return Token(ch);        // let each character represent itself
+        return Token{ ch }; // let each character represent itself
     case '.':
     case '0': case '1': case '2': case '3': case '4':
     case '5': case '6': case '7': case '8': case '9':
```

Comment:
fix the brackets and indentation. something like:

case 'x':
{
	do something;
	do another;
}

Reply:
These changes have been made

Reply:
after the opening '{', insert a new line. see teams-bjarne dance-post for an example